### DISCUSSION TITLE
including more than one implicit many-to-many relation

### DISCUSSION THREAD

**Discussion by ranran2121** [prisma/prisma#16983] [Q&A] [answered]
Hi,
i have the following (simplified) schema
```
model Profile {
  id Int @id @default(autoincrement())
  name String
  jobExpectations JobExpectation[]
  otherExpectations OtherExpectation[]
}

model JobExpectation {
  id Int @id @default(autoincrement())
  name String @unique
  profile Profile[]
}

model OtherExpectation {
  id Int @id @default(autoincrement())
  name String @unique
  profile Profile[]
}

```
the table Profile has an implicit many-to-many relation to the JobExpectation table and another implicit many-to-many relation to the OtherExpectation table

I use the following query to retrieve the profile information with the JobExpectations
```
const profile = await prisma.profile.findUnique({
      where: { userId: user.id },
      include: {
        jobExpectations: true,
      },
    });
```
however I need to retrieve also the otherExpectations array but this syntax does NOT work
```
const profile = await prisma.profile.findUnique({
      where: { userId: user.id },
      include: {
        jobExpectations: true,
       otherExpectations: true,
      },
    });
```

Can somebody suggest me how to retrieve BOTH the arrays in one query?

**Comment by nurul3101** [comment, marked as answer]
Hello @ranran2121 👋 

It is possible to retrieve both the relation fields i.e. `jobExpectations` and `otherExpectations`, what error are you getting?
Here's the query that I tried and it works as expected.
```typescript
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

async function main() {
  const profile = await prisma.profile.create({
    data: {
      id: 1,
      name: 'John Doe',
      jobExpectations: {
        create: {
          name: 'Job Expectation 1',
        },
      },
      otherExpectations: {
        create: {
          name: 'Other Expectation 1',
        },
      },
    },
  });

  const profileResult = await prisma.profile.findUnique({
    where: { id: 1 },
    include: {
      jobExpectations: true,
      otherExpectations: true,
    },
  });

  console.log(JSON.stringify(profileResult, null, 2));
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

Result:

<img width="501" alt="Screenshot 2022-12-22 at 10 08 39 PM" src="https://user-images.githubusercontent.com/13049676/209180848-da20008c-08c4-4b23-886c-aee078abdc28.png">

**Reply by ranran2121** [reply]
mmmm, that is weird...
I did not get any error, just was not getting the user.otherExpectations array. I tried it again and now it works

**Reply by nurul3101** [reply]
I am glad to hear that the issue is resolved now 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
